crates/nu-cli/src/repl.rs

```diff
@@ -357,9 +387,19 @@ pub fn evaluate_repl(
 
         start_time = std::time::Instant::now();
         let config = &engine_state.get_config().clone();
-        prompt_update::update_prompt(config, engine_state, stack, &mut nu_prompt);
-        let transient_prompt =
-            prompt_update::make_transient_prompt(config, engine_state, stack, &nu_prompt);
+        prompt_update::update_prompt(
+            config,
+            engine_state,
+            &mut Stack::with_parent(stack.clone()),
+            &mut nu_prompt,
+        );
+        // STACK-REFERENCE 4
```

Comment:
This does clone the `Arc`, but it's a temporary reference and will be dropped right after `make_transient_prompt`. Note how `update_prompt` above doesn't need to be marked as a `STACK-REFERENCE`.